src/Spice86.Core/Emulator/OperatingSystem/DosPathResolver.cs

```diff
@@ -0,0 +1,232 @@
+            convertParentOnly) {
+            // file exists or root reached, no need to go further. Path found.
+            return dosPath;
+        }
+
+        string? parent = ToCaseSensitivePath(parentDir, convertParentOnly);
+        if (parent == null) {
+            // End of recursion, root reached
+            return null;
+        }
+
+        // Now that parent is for sure on the host file system, let's find the actual path
+        try {
+            string? fileNameOnFileSystem = TryGetFullHostFileName(dosPath);
+            if (!string.IsNullOrWhiteSpace(fileNameOnFileSystem)) {
+                return fileNameOnFileSystem;
+            }
+            string searchPattern = Path.GetFileName(hostPath);
+            if (Directory.Exists(parent)) {
+                return
+                    Directory.GetDirectories(parent, searchPattern)
+                    .Concat(
+                    Directory.GetFiles(parent, searchPattern))
+                    .FirstOrDefault();
+            }
+        } catch (IOException e) {
+            e.Demystify();
+            if (_loggerService.IsEnabled(Serilog.Events.LogEventLevel.Warning)) {
+                _loggerService.Warning(e, "Error while checking file {CaseInsensitivePath}: {Exception}", dosPath, e);
+            }
+        }
+
+        return null;
+    }
+    
+    /// <inheritdoc />
+    public string? ToHostCaseSensitiveFullName(string dosPath, bool convertParentOnly) {
+        string fileName = PrefixWithHostDirectory(dosPath);
+        if (!convertParentOnly) {
+            string? caseSensitivePath = ToCaseSensitivePath(fileName, convertParentOnly);
+            return caseSensitivePath;
+        }
+        
+        string? parent = ToCaseSensitivePath(fileName, convertParentOnly);
+        if (string.IsNullOrWhiteSpace(parent)) {
+            return null;
+        }
+        
+        return ConvertUtils.ToSlashPath(parent);
+    }
+
+    /// <inheritdoc />
+    public string PrefixWithHostDirectory(string dosPath) {
+        string path = dosPath;
+
+        if(string.IsNullOrWhiteSpace(path)) {
+            return path;
+        }
+
+        if(IsPathRooted(path)) {
+            path = Path.Combine(DriveMap[path[0]].FullName, path[3..]);
+        }
+        else if (StartsWithDosDrive(dosPath)) {
+            path = Path.Combine(CurrentHostDirectory, path[2..]);
+        } else {
+            path = Path.Combine(CurrentHostDirectory, path);
+        }
+
+        return ConvertUtils.ToSlashPath(path);
+    }
+
+    /// <summary>
+    /// All the possible DOS drive letters
+    /// </summary>
+    private static IEnumerable<char> DriveLetters => "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+    private bool StartsWithDosDrive(string path) => 
+        path.Length >= 2 &&
+        DriveLetters.Contains(char.ToUpperInvariant(path[0])) &&
+        path[1] == ':';
+
+    private bool IsPathRooted(string path) => 
+        path.Length >= 3 &&
+        StartsWithDosDrive(path) &&
```

Comment:
But, paths that start with a slash are also rooted aren't they?
I mean, if I'm in `d:\data\docs` and type `cd \Tools` then I'm in `d:\tools`

Reply:
Paths that start with '\' are now considered to be rooted.